### PR TITLE
[CUMULUS-2364] Remove log2elasticsearch

### PR DESCRIPTION
**Pull request by bt6453**
**Summary:** Summary of changes

Addresses [CUMULUS-2364: Remove log2elasticsearch](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2364)

## Changes

* Remove code related to log2elasticsearch
* Update terraform code to no longer deploy log modules if not set up for Metrics

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: tf-modules/log-subscription-filter/main.tf [hunk 1]
```diff
@@ -6,7 +6,3 @@ terraform {
     }
   }
 }
-
-locals {
-  destination_arn = var.log_destination_arn != null ? var.log_destination_arn : var.log2elasticsearch_lambda_function_arn
```
Comment (nemreid) [inline tf-modules/log-subscription-filter/main.tf:11]: So what happens when `var.log_destination_arn` is null now? Does the deployment blow up?

Comment (bt6453) [reply tf-modules/log-subscription-filter/main.tf:11]: The module won't even be deployed if there is no `log_destination_arn`. See: https://github.com/nasa/cumulus/blob/012de8a87ed6cfc4cfc98b13fc7151c6dcfdbca3/tf-modules/cumulus/log_subscription_filter.tf#L2

File: tf-modules/cumulus/log_subscription_filter.tf [hunk 2]
```diff
@@ -1,10 +1,10 @@
 module "log_subscription_filter" {
+  count = var.log_destination_arn != null ? 1 : 0
```
Comment (Jkovarik) [inline tf-modules/cumulus/log_subscription_filter.tf:2]: Terraform 13 is 💯

**Review by Jkovarik — CHANGES_REQUESTED** [review]
This all looks good!!!   I reviewed the code, as well as deployed with and and without the log_destination_arn set, and confirmed with it set (targeting a test lambda) that it triggered.  

 A couple super minor bits to tidy up/comments to address (I added an amended PR with suggestions found while reviewing here:  https://github.com/nasa/cumulus/pull/2100)

File: tf-modules/cumulus/log_subscription_filter.tf (hunk 2 above)
Comment (bt6453) [reply tf-modules/cumulus/log_subscription_filter.tf:2]: Saved me so much copy and pasting

File: tf-modules/log-subscription-filter/variables.tf [hunk 3]
```diff
@@ -9,11 +9,6 @@ variable "log_destination_arn" {
   description = "A shared AWS:Log:Destination that receives logs from log_groups"
```
[the remark this answers is not in the document]
Comment (bt6453) [reply tf-modules/log-subscription-filter/variables.tf:8]: Done

**Comment by bt6453** [comment]
PR updated with Jonathan's suggestions

**Review by Jkovarik — APPROVED** [review]
Looks good!  Nice work.

**Pull request merged**